src/common/groupedOrdersTable.component.tsx

```diff
@@ -0,0 +1,261 @@
+import { formatDate, parseDate, usePagination } from "@openmrs/esm-framework";
+import { GroupedOrdersTableProps } from "./commonProps.resource";
+import { useSearchGroupedResults } from "../hooks/useSearchGroupedResults";
+import {
+  Table,
+  TableHead,
+  TableRow,
+  TableHeader,
+  TableBody,
+  TableExpandRow,
+  TableExpandedRow,
+  TableExpandHeader,
+  TableCell,
+  DataTable,
+  Pagination,
+  TableContainer,
+  TableToolbar,
+  TableToolbarContent,
+  TableToolbarSearch,
+  Layer,
+  Dropdown,
+  DatePicker,
+  DatePickerInput,
+} from "@carbon/react";
+import ListOrderDetails from "./listOrderDetails.component";
+import Overlay from "../components/overlay/overlay.component";
+
+//  render Grouped by patient Orders in procedures app
+const GroupedOrdersTable: React.FC<GroupedOrdersTableProps> = (props) => {
+  const workListEntries = props.orders;
+  const { t } = useTranslation();
+  const [currentPageSize, setCurrentPageSize] = useState<number>(10);
+  const [searchString, setSearchString] = useState<string>("");
+  const [activatedOnOrAfterDate, setActivatedOnOrAfterDate] = useState("");
+
+  const OrderStatuses = [
+    "All",
+    "RECEIVED",
+    "IN_PROGRESS",
+    "COMPLETED",
+    "EXCEPTION",
+    "ON_HOLD",
+    "DECLINED",
+  ];
+
+  const [filter, setFilter] = useState<
+    | "All"
+    | "EXCEPTION"
+    | "RECEIVED"
+    | "COMPLETED"
+    | "IN_PROGRESS"
+    | "ON_HOLD"
+    | "DECLINED"
+  >("All");
+
+  const filteredEntries = useMemo(() => {
+    if (!filter || filter == "All") {
+      return workListEntries;
+    }
+
+    if (filter) {
+      return workListEntries?.filter(
+        (order) => order.fulfillerStatus === filter
+      );
+    }
+
+    return workListEntries;
+  }, [filter, workListEntries]);
+
+  const handleOrderStatusChange = ({ selectedItem }) => setFilter(selectedItem);
+
+  function groupOrdersById(orders) {
+    if (orders && orders.length > 0) {
+      const groupedOrders = orders.reduce((acc, item) => {
+        if (!acc[item.patient.uuid]) {
+          acc[item.patient.uuid] = [];
+        }
+        acc[item.patient.uuid].push(item);
+        return acc;
+      }, {});
+
+      // Convert the result to an array of objects with patientId and orders
+      return Object.keys(groupedOrders).map((patientId) => ({
+        patientId: patientId,
+        orders: groupedOrders[patientId],
+      }));
+    } else {
+      return [];
+    }
+  }
+  const groupedOrdersByPatient = groupOrdersById(filteredEntries);
+  const searchResults = useSearchGroupedResults(
+    groupedOrdersByPatient,
+    searchString
+  );
+  const {
+    goTo,
+    results: paginatedResults,
+    currentPage,
+  } = usePagination(searchResults, currentPageSize);
+
+  const pageSizes = [10, 20, 30, 40, 50];
+
+  const rowsdata = useMemo(() => {
+    return paginatedResults.map((patient) => ({
+      id: patient.patientId,
+      patientname: patient.orders[0].patient?.display?.split("-")[1],
```

Comment:
```suggestion
      patientName: patient.orders[0].patient?.display?.split("-")[1],
```
please use camelCase